dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -45,140 +54,147 @@
-
-		            if(bundleName.trim().length()>0) {
-					    // save bundle if it doesn't exists
-		                Bundle foundBundle = APILocator.getBundleAPI().getBundleById( bundleFolder );
-		                if ( foundBundle == null || foundBundle.getId() == null ) {
-		                    Bundle bundle = new Bundle();
-							bundle.setId(bundleFolder);
-							bundle.setName(bundleName);
-							bundle.setPublishDate(Calendar.getInstance().getTime());
-							bundle.setOwner(APILocator.getUserAPI().getSystemUser().getUserId());
-							bundle.setForcePush(forcePush);
-		                    APILocator.getBundleAPI().saveBundle(bundle);
-					    }
-					}
-
-					//Write file on FS
-					FileUtil.writeToFile(bundleStream, bundlePath+fileName);
-
-					//Start thread
-					if(!status.getStatus().equals(Status.PUBLISHING_BUNDLE)) {
-						new Thread(new PublishThread(fileName, groupId, endpointId, status)).start();
-					}
-
-					HibernateUtil.commitTransaction();
-
-					return Response.status(HttpStatus.SC_OK).build();
-				} catch (NumberFormatException e) {
-				    try {
-		                HibernateUtil.rollbackTransaction();
-		            } catch (DotHibernateException e1) {
-		                Logger.error(this, "error rollback",e1);
-		            }
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				} catch (Exception e) {
-				    try {
-		                HibernateUtil.rollbackTransaction();
-		            } catch (DotHibernateException e1) {
-		                Logger.error(this, "error rollback",e1);
-		            }
-					Logger.error(PublisherQueueJob.class, "Error caused by remote call of: "+remoteIP);
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				}
-				finally {
-				    try {
-		                HibernateUtil.closeSession();
-		            } catch (DotHibernateException e) {
-		                Logger.error(this, "error close session",e);
-		            }
+			return new WebResource.InitBuilder().
+					rejectWhenNoUser(false). // it would be a soft validation so not reject
+					requestAndResponse(
+					new HttpHeaderHandlerHttpServletRequestWrapper(request,
+							CollectionsUtils.map(
+									"Authorization", (name, value) -> // if the authorization is set, uses it, otherwise try with the secret (could be a jwt)
+											UtilMethods.isSet(value) ? value : JsonWebTokenAuthCredentialProcessor.BEARER + authTokenDigest
+							)), response).init();
+		}catch (Exception e) {
+    		return null;
+		}
+
+	}
+
+
+	@WrapInTransaction
+	private void publishBundle(final String fileName,
+								  final String groupId,
+								  final String endpointId,
+								  final String bundleName,
+								  final boolean forcePush,
+								  final HttpServletRequest request,
+								  final String remoteIP,
+								  final PublishingEndPoint sendingEndPointByAddress) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5146761)